src/components/inputs/ColorSelect2/ColorSelect2.test.tsx

```diff
@@ -0,0 +1,225 @@
+    await userEvent.click(trigger);
+    const picker = screen.getByLabelText('color picker');
+    expect(picker).toBeInTheDocument();
+  });
+
+  it('Set ColorSelect2 color using value prop', async () => {
+    renderWithThemeProvider(<ColorSelect2 value={TEST_COLOR} />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    const selectedColor = screen.getByLabelText('color preview');
+    const selectedColorValue = selectedColor.getAttribute('color');
+    expect(selectedColorValue).toBe(TEST_COLOR);
+  });
+
+  it('Change ColorSelect2 color using input', async () => {
+    renderWithThemeProvider(<ColorSelect2 />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    fireEvent.change(input, { target: { value: TEST_COLOR } });
+
+    const selectedColor = screen.getByLabelText('color preview');
+    const selectedColorValue = selectedColor.getAttribute('color');
+    expect(selectedColorValue).toBe(TEST_COLOR);
+  });
+
+  it('Fires onChange callback', async () => {
+    const mockFn = jest.fn();
+
+    renderWithThemeProvider(<ColorSelect2 onChange={mockFn} />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    fireEvent.change(input, { target: { value: TEST_COLOR } });
+
+    expect(mockFn).toBeCalledWith(TEST_COLOR);
+  });
+
+  it('Fires onClose callback', async () => {
+    const mockFn = jest.fn();
+
+    renderWithThemeProvider(<ColorSelect2 onClose={mockFn} />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+    await userEvent.click(input); // Another click to close the picker
+
+    expect(mockFn).toBeCalled();
+  });
+
+  it('Resets ColorSelect2 color using reset button', async () => {
+    const reset = { color: '#FFFFFF', label: 'My Reset' };
+
+    renderWithThemeProvider(<ColorSelect2 reset={reset} />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    fireEvent.change(input, { target: { value: TEST_COLOR } });
+
+    const resetButton = screen.getByLabelText('reset');
+    await userEvent.click(resetButton);
+
+    const selectedColor = screen.getByLabelText('color preview');
+    const selectedColorValue = selectedColor.getAttribute('color');
+    expect(selectedColorValue).toBe(reset.color);
+  });
+
+  it('Change size', async () => {
+    const { rerender } = renderWithThemeProvider(
+      <ColorSelect2 size="sm" />
+    );
+
+    const label = screen.getByLabelText('color');
+
+    expect(label).toHaveStyle({ 'font-size': '0.75rem' });
+
+    rerender(<ColorSelect2 size="md" />);
+    expect(label).toHaveStyle({ 'font-size': '1rem' });
+
+    rerender(<ColorSelect2 size="lg" />);
+    expect(label).toHaveStyle({ 'font-size': '1.5rem' });
+
+    rerender(<ColorSelect2 size="xl" />);
+    expect(label).toHaveStyle({ 'font-size': '1.75rem' });
+  });
+
+  // it('Make ColorSelect2 required', async () => {
```

Comment:
Instead of commenting this out, let's skip it using `xit` instead while putting a note on why it's being skipped. It'll be a reminder when the tests are run that it needs to be addressed.

Reply:
done